freeipa/src/main/java/com/sequenceiq/freeipa/controller/UserV1Controller.java

```diff
@@ -109,6 +114,14 @@ public SyncOperationStatus getSyncOperationStatus(@NotNull String operationId) {
                 operationService.getOperationForAccountIdAndOperationId(accountId, operationId));
     }
 
+    @Override
+    @CheckPermissionByAccount(action = AuthorizationResourceAction.ENVIRONMENT_READ)
+    public EnvironmentUserSyncState getUserSyncState(@NotNull String environmentCrn) {
+        LOGGER.debug("getUserSyncState() requested for environment '{}'", environmentCrn);
+
+        return environmentUserSyncStateCalculator.calculateEnvironmentUserSyncState(Crn.safeFromString(environmentCrn));
```

Comment:
I disagree with your suggestion here. I've updated the PR to match current behavior of the other APIs, but will start an email discussion about the API usage of the actor crn since this issue is larger than this PR.

Reply:
sure, @horadla23 knows this stuff better

Reply:
@handavid as i mentioned, with `@TenantAwareParam` annotation, you can get the actual account id from `ThreadBasedUserCrnProvider`

Reply:
@horadla23  I looked at the code to the `InternalCrnModifier` and I find it strange that it changes the CRN stored in the `ThreadBasedUserCrnProvider` into a mixture of the original user crn and the account from the annotated field. I don't think this is desirable because you may end up with an invalid actor that would fail authz on subsequent API calls.

Instead, I think I need the changes you have in CB-7104 and to use the `@CheckPermissionByResourceCrn` annotation instead since the request includes an environment crn.

Reply:
@handavid we are doing that only in case of internal actpr and we do not call ums with internal actor since everything is allowed with that, so it is working, other team:s service can call environment service this way without problem. You can still use @CheckPermissionByResourceCrn, as you can see from example from my email